src/ts/components/errormessageoverlay.ts

```diff
@@ -4,9 +4,10 @@ import {UIInstanceManager} from '../uimanager';
 import ErrorEvent = bitmovin.PlayerAPI.ErrorEvent;
 import {TvNoiseCanvas} from './tvnoisecanvas';
 import PlayerEvent = bitmovin.PlayerAPI.PlayerEvent;
+import {ErrorEventUtil} from '../erroreventutil';
 
 export interface ErrorMessageTranslator {
-  (error: ErrorEvent): string;
+  (description: string, error: ErrorEvent): string;
```

Comment:
The event is more important so it should be the first parameter. Also the description is optional because there's not a description for every code. Actually I think this is not a nice design as the `ErrorMessageTranslator` should output the "description" and not receive it as input. Also the name `description` seems a little off, `errorMessage` would be more fitting.

Reply:
Yes but we where thinking about using the callback (`ErrorMessageTranslator`) to 'format' the description without the need to define an own.

Reply:
Hm maybe possible when accessing the `defaultErrorMessagesMapping` within the callback by yourself.

Reply:
Exactly. Implement is as an `ErrorMessageTranslator`, within it you can do whatever you want, i.e. processing placeholders. Then use it as fallback as described in the comment below.